example/test_driver/sdk_vendor_test.dart

```diff
@@ -0,0 +1,195 @@
+          completes);
+    });
+
+    test("Get enabled purpose ids without initialization", () async {
+      String onReadyReceived = await driver.requestData("isOnReadyReceived");
+      assert(onReadyReceived == "false");
+
+      await driver.tap(enabledVendorIdsBtnFinder);
+      expect(
+          driver.getText(find.byValueKey("nativeResponse_getEnabledVendorIds")).then(
+              (field) => expect(field, contains("Native message: Failed: \'Didomi SDK is not ready. Use the onReady callback to access this method.\'."))),
+          completes);
+    });
+
+    test("Get required purpose ids without initialization", () async {
+      String onReadyReceived = await driver.requestData("isOnReadyReceived");
+      assert(onReadyReceived == "false");
+
+      await driver.tap(requiredVendorIdsBtnFinder);
+      expect(
+          driver.getText(find.byValueKey("nativeResponse_getRequiredVendorIds")).then(
+              (field) => expect(field, contains("Native message: Failed: \'Didomi SDK is not ready. Use the onReady callback to access this method.\'."))),
+          completes);
+    });
+
+    /*
+     * With initialization
+     */
+
+    test("Get disabled purpose ids with initialization", () async {
+      await driver.tap(initializeBtnFinder);
+      await Future.delayed(Duration(seconds: 1));
+
+      String onReadyReceived = await driver.requestData("isOnReadyReceived");
+      assert(onReadyReceived == "true");
+
+      await driver.tap(disabledVendorIdsBtnFinder);
+      expect(
+          driver
+              .getText(find.byValueKey("nativeResponse_getDisabledVendorIds"))
+              .then((field) => expect(field, contains("Native message: Disabled Vendors list is empty."))),
+          completes);
+    });
+
+    test("Get enabled purpose ids with initialization", () async {
+      String onReadyReceived = await driver.requestData("isOnReadyReceived");
+      assert(onReadyReceived == "true");
+
+      await driver.tap(enabledVendorIdsBtnFinder);
+      expect(
+          driver
+              .getText(find.byValueKey("nativeResponse_getEnabledVendorIds"))
+              .then((field) => expect(field, contains("Native message: Enabled Vendors list is empty."))),
+          completes);
+    });
+
+    test("Get required purpose ids with initialization", () async {
+      String onReadyReceived = await driver.requestData("isOnReadyReceived");
+      assert(onReadyReceived == "true");
+
+      await driver.tap(requiredVendorIdsBtnFinder);
+      expect(
+          driver.getText(find.byValueKey("nativeResponse_getRequiredVendorIds")).then((field) => expect(
+              field,
+              contains(
+                  "Native message: Required Vendors: 164,284,402,413,91,c:ab-tasty,c:adjust-JZUjXFCR,c:contentsquare,c:facebook-dmtCpRLw,c:omniture-adobe-analytics,c:pinterest,c:snapchat-b8Vdit6J,c:tiktok-KZAUQLZ9,c:tradedoubler,google."))),
```

Comment:
Should we check all the vendors? As we test on a demo notice, the list could be changed, so maybe checking a subset would be enough

Reply:
The goal is to check the response with a selected configuration.
This test use the reduced Initialize widget. This one is only use for ui tests and doesn't allow to change the config.
I think it should work well as it is and we still can use the sample with other configurations without breaking those tests